tedana/tests/test_metrics.py

```diff
@@ -207,15 +207,41 @@ def test_smoke_calculate_f_maps():
     mixing = np.random.random((n_volumes, n_components))
     adaptive_mask = np.random.randint(1, n_echos + 1, size=n_voxels)
     tes = np.array([15, 25, 35, 45, 55])
+    f_t2_maps_orig, f_s0_maps_orig, _, _ = dependence.calculate_f_maps(
+        data_cat=data_cat,
+        z_maps=z_maps,
+        mixing=mixing,
+        adaptive_mask=adaptive_mask,
+        tes=tes,
+        f_max=500,
+    )
+    assert f_t2_maps_orig.shape == f_s0_maps_orig.shape == (n_voxels, n_components)
+
+    # rerunning with echo_dof=3
     f_t2_maps, f_s0_maps, _, _ = dependence.calculate_f_maps(
         data_cat=data_cat,
         z_maps=z_maps,
         mixing=mixing,
         adaptive_mask=adaptive_mask,
         tes=tes,
+        echo_dof=3,
         f_max=500,
     )
     assert f_t2_maps.shape == f_s0_maps.shape == (n_voxels, n_components)
+    # When echo_dof < the number of echoes, then f_maps_orig should have the same or larger values
+    assert np.round(np.min(f_t2_maps_orig - f_t2_maps), decimals=3) == np.round(0.0, decimals=3)
+    assert np.round(np.min(f_s0_maps_orig - f_s0_maps), decimals=3) == np.round(0.0, decimals=3)
+    # When echo_dof==3, there are 5 good echoes, and f_maps_orig>0
+    # then f_maps_orig should always be larger than f_maps with fewer DOF
+    echo5_mask = np.logical_and(
+        np.logical_and(
+            np.tile(adaptive_mask == 5, (50, 1)).T,
+            f_t2_maps_orig > 0.2,
+        ),
+        f_t2_maps_orig < 499,
+    )
+    assert np.min(f_t2_maps_orig[echo5_mask] - f_t2_maps[echo5_mask]) > 0.0
+    assert np.min(f_s0_maps_orig[echo5_mask] - f_s0_maps[echo5_mask]) > 0.0
```

Comment:
```suggestion
    t2_echo5_mask = np.logical_and(
        np.logical_and(
            np.tile(adaptive_mask == 5, (n_components, 1)).T,
            f_t2_maps_orig > 0.2,
        ),
        f_t2_maps_orig < 499,
    )
    assert np.min(f_t2_maps_orig[t2_echo5_mask] - f_t2_maps[t2_echo5_mask]) > 0.0
    s0_echo5_mask = np.logical_and(
        np.logical_and(
            np.tile(adaptive_mask == 5, (n_components, 1)).T,
            f_s0_maps_orig > 0.2,
        ),
        f_s0_maps_orig < 499,
    )
    assert np.min(f_s0_maps_orig[s0_echo5_mask] - f_s0_maps[s0_echo5_mask]) > 0.0
```
Realized that f_t2_maps was fine, but f_s0_maps was not. Frustrating that this seems to reliably fail here, but reliably passes on my computer.